src/compiler/checker.ts

```diff
@@ -20422,6 +20422,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 const newMapper = createTypeMapper(root.outerTypeParameters, typeArguments);
                 const checkType = root.checkType;
                 let distributionType = root.isDistributive ? getReducedType(getMappedType(checkType, newMapper)) : undefined;
+                if (distributionType && isNoInferType(distributionType)) {

```

Comment:
I have 2 doubts here:
1. This still doesn't solve the issue with `A | NoInfer<B | C>`. To solve that, it feels like `mapTypeWithAlias` and `mapType` should learn how to deal with those NoInfer-wrapped unions
2. Should the conditional type's result be "repacked" into `NoInfer` back if needed? 🤔